### DISCUSSION TITLE
TexTeller改进建议

### DISCUSSION THREAD

**Discussion by circlestarzero** [OleehyO/TexTeller#62] [feedback]
TexTeller在标准公式上效果不错, 但是会有非标准语法(mathjax不支持的, 需要用到特定的latex宏包, 推测部分数据生成方式是参考nougat 从arxiv 使用latexml直接提取的),  期待进一步完善
latex语法清洗上可以进一步改进, 例如\bm, \newline等, 考虑适配mathjax

**Comment by circlestarzero** [comment]
也可以考虑加个正则后处理, 来兼容 mathjax 语法

**Comment by OleehyO** [comment]
texteller目前确实是通过正则来处理非标准语法，只不过是转化到katex语法上，后续有时间的话应该会考虑再兼容mathjax🤔

**Comment by circlestarzero** [comment]
mathjax 的语法支持比 katex 好, 未来希望支持一下mathjax

**Comment by Sanster** [comment]
> TexTeller在标准公式上效果不错, 但是会有非标准语法(mathjax不支持的, 需要用到特定的latex宏包, 推测部分数据生成方式是参考nougat 从arxiv 使用latexml直接提取的), 期待进一步完善 latex语法清洗上可以进一步改进, 例如\bm, \newline等, 考虑适配mathjax

@circlestarzero 你好，想请教下关于你说的”latex语法清洗上可以进一步改进, 例如\bm, \newline等, 考虑适配mathjax“，这个过程是不是类似于 [latex-ocr](https://github.com/lukas-blecher/LaTeX-OCR?tab=readme-ov-file#dataset-requirements) 中借助 katex 的 AST 进行 normalizing？
